src/test/java/onseo/jerseytest/MockSampleTest.java

```diff
@@ -31,25 +37,120 @@
     private SummaryService summaryService;
 
     @Before
-    public void setup()
-    {
-        when(toDoEntityPlaceholderClient.getEntity(Matchers.anyInt(), any(), any())).thenReturn(generateStubEntity());
-        summaryService = new SummaryService(toDoEntityPlaceholderClient, commentEntityPlaceholderClientMock, userEntityPlaceholderClientMock);
+    public void setup() throws ExecutionException, InterruptedException {
+        CompletableFuture<ToDoEntity> todoTask = CompletableFuture.supplyAsync(this::generateToDoStubEntity);
+        CompletableFuture<CommentEntity> commentTask = CompletableFuture.supplyAsync(this::generateCommentStubEntity);
+        CompletableFuture<UserEntity> userTask = CompletableFuture.supplyAsync(this::generateUserStubEntity);
+
+        when(toDoEntityPlaceholderClientMock.getEntity(Matchers.anyInt(), any(), any())).thenReturn(todoTask.get());
+        when(commentEntityPlaceholderClientMock.getEntity(Matchers.anyInt(), any(), any())).thenReturn(commentTask.get());
+        when(userEntityPlaceholderClientMock.getEntity(Matchers.anyInt(), any(), any())).thenReturn(userTask.get());
+        summaryService = new SummaryService(toDoEntityPlaceholderClientMock, commentEntityPlaceholderClientMock, userEntityPlaceholderClientMock);
     }
 
     @Test
     public void urlToSampleString() {
-        System.out.println(summaryService.getToDo(111));
+        System.out.println(summaryService.getToDo(10));
     }
 
-    private ToDoEntity generateStubEntity()
+    @Test
+    public void urlToSampleGetSummary() throws InterruptedException {
+
+        System.out.println(summaryService.getEntityAsync(10, "/todos/", ToDoEntity.class));
+        System.out.println(summaryService.getEntityAsync(8, "/comments/", CommentEntity.class));
+        System.out.println(summaryService.getEntityAsync(5, "/users/", UserEntity.class));
+        System.out.println(summaryService.getSummaryAsync(5));
+    }
+
+    private ToDoEntity generateToDoStubEntity()
     {
+        log.info("Generating stub ToDo entity...");
         ToDoEntity result = new ToDoEntity();
         result.setId(100);
         result.setUserId(200);
         result.setTitle("Some temporary title.");
         result.setCompleted(true);
 
+        try
+        {
+            log.info("ToDo thread ({}) goes to sleep", Thread.currentThread().getName());
+            Thread.sleep(500);
```

Comment:
add delay in args of generateXXXXX() methods or consts in order to manage in one place